common/tools/turborepo-remote-cache/src/index.ts

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import "dotenv/config";
+import Fastify from "fastify";
+import { badRequest, notFound } from "@hapi/boom";
+import { DefaultAzureCredential } from "@azure/identity";
+import { BlobServiceClient, ContainerClient } from "@azure/storage-blob";
+import { join } from "node:path";
+import { PassThrough, Readable } from "node:stream";
+import { authenticateToken, authenticateTeamId } from "./auth.js";
+
+const MAX_FILE_SIZE = 50 * 1024 * 1024; // 50MB
+
+const logger = {
+  pretty: {
+    transport: {
+      target: "pino-pretty",
+      options: {
+        translateTime: "HH:MM:ss Z",
+        ignore: "pid,hostname",
+      },
+    },
+  },
+  basic: {
+    level: process.env.CACHE_LOG_LEVEL || "info",
+  },
+};
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let pinoPretty: any | undefined = undefined;
+try {
+  pinoPretty = await import("pino-pretty");
+} catch {
+  // no-op
+}
+const fastify = Fastify({
+  logger: pinoPretty ? logger.pretty : logger.basic,
+  exposeHeadRoutes: false,
+  bodyLimit: MAX_FILE_SIZE,
+});
+
+fastify.addContentTypeParser(
+  "application/octet-stream",
+  { parseAs: "buffer", bodyLimit: MAX_FILE_SIZE },
+  (_req, body, done) => {
+    done(null, body);
+  },
+);
+
+fastify.addHook("preHandler", authenticateTeamId);
+
+const port = (process.env.AZURE_CACHE_PORT || 3000) as number;
+const packageVersion = process.env.PACKAGE_VERSION || "1.0.0";
+
+let containerClient: ContainerClient;
+
+if (process.env.AZURE_CONTAINER_SAS_URL) {
```

Comment:
Is this in the bicep file?

Reply:
No, this offers a way to test/run the cache server without using DefaultAzureCredential. We will not have or use it in the deployment.